### PR TITLE
[GH-1013] Fix visibility of clear button for select option

### LINKED ISSUE #1013: Select option clear button display is inconsistent
**Steps to reproduce the behavior:**
1. Open a table with select and multi-select properties
2. Notice that the select options have a clear (X) button on them
3. Notice that the multi-select options do not, unless editing
4. Open a card, notice a similar inconsistency in the X button display

**Expected behavior:**
Select options should not show a clear button when not editing, across all views.

**Screenshots:**
Table select option

When editing a multi-select property, the clear buttons appear:
Table multi-select edit

**Platform:**
 - v0.9.0 RC5 on Community
 - Chrome 91 on macOS

### PR DESCRIPTION
**Pull request by kamre**
#### Summary
Show clear button for (multi) select property only when it is being edited:
- set open state to `false` on `blur` event for (multi) select property
- files for `MultiSelectProperty` component moved into separate folder
- extracted new `SelectProperty` component from the code inside `PropertyValueElement`
- unit test for new component added
- minor improvements for styles for `Label` component

Here is how it looks for the table:
Select_MultiSelect_Table

For the card dialog:
Select_MultiSelect_CardDialog

#### Ticket Link
Fixes #1013

### CODE REVIEW

File: webapp/src/components/properties/select/select.test.tsx [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+                onCreate={onCreate}
+            />,
+        ))
+
+        userEvent.click(screen.getByTestId(nonEditableSelectTestId))
+        userEvent.type(screen.getByRole('textbox', {name: /value selector/i}), `${newOption}{enter}`)
+
+        expect(onCreate).toHaveBeenCalledWith(newOption)
+    })
+})
```
Comment (sbishel) [inline webapp/src/components/properties/select/select.test.tsx:193]: Nice tests and coverage...

File: webapp/src/components/properties/select/select.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            onChange={(value) => props.onChange(value as string)}
+            onChangeColor={props.onChangeColor}
+            onDeleteOption={props.onDeleteOption}
+            onDeleteValue={props.onDeleteValue}
+            onBlur={() => setOpen(false)}
+        />
+    )
+})
+
+export default SelectProperty
```
Comment (sbishel) [inline webapp/src/components/properties/select/select.tsx:61]: I like this enhancement.

**Review by sbishel — CHANGES_REQUESTED** [review]
Looks awesome. Thanks @kamre Looks like some tests failed after I sync'ed to main. Could you update and I'll get this merged.

**Comment by kamre** [comment]
@sbishel updated jest snapshot.

**Review by sbishel — APPROVED** [review]

**Pull request merged**